src/game/client/worldheightmap.cpp

```diff
@@ -29,21 +45,2359 @@ class GDIFileStream : public InputStream
+        m_drawHeightY = m_height;
+    }
+
+    g_theSidesList->Validate_Sides();
+    Setup_Alpha_Tiles();
+}
+
+WorldHeightMap::~WorldHeightMap()
+{
+    if (m_data) {
+        delete[] m_data;
+        m_data = nullptr;
+    }
+
+    if (m_tileNdxes) {
+        delete[] m_tileNdxes;
+        m_tileNdxes = nullptr;
+    }
+
+    if (m_blendTileNdxes) {
+        delete[] m_blendTileNdxes;
+        m_blendTileNdxes = nullptr;
+    }
+
+    if (m_extraBlendTileNdxes) {
+        delete[] m_extraBlendTileNdxes;
+        m_extraBlendTileNdxes = nullptr;
+    }
+
+    if (m_cliffInfoNdxes) {
+        delete[] m_cliffInfoNdxes;
+        m_cliffInfoNdxes = nullptr;
+    }
+
+    if (m_cellFlipState) {
+        delete[] m_cellFlipState;
+        m_cellFlipState = nullptr;
+    }
+
+    if (m_seismicUpdateFlag) {
+        delete[] m_seismicUpdateFlag;
+        m_seismicUpdateFlag = nullptr;
+    }
+
+    if (m_seismicZVelocities) {
+        delete[] m_seismicZVelocities;
+        m_seismicZVelocities = nullptr;
+    }
+
+    if (m_cliffState) {
+        delete[] m_cliffState;
+        m_cliffState = nullptr;
+    }
+
+    for (int i = 0; i < TILE_COUNT; i++) {
+        Ref_Ptr_Release(m_sourceTiles[i]);
+        Ref_Ptr_Release(m_edgeTiles[i]);
+    }
+
+    for (int i = 0; i < 12; i++) {
+        Ref_Ptr_Release(s_alphaTiles[i]);
+    }
+
+    Ref_Ptr_Release(m_terrainTex);
+    Ref_Ptr_Release(m_alphaTerrainTex);
+}
+
+void WorldHeightMap::Free_List_Of_Map_Objects()
+{
+    if (MapObject::s_theMapObjectListPtr) {
+        MapObject::s_theMapObjectListPtr->Delete_Instance();
+        MapObject::s_theMapObjectListPtr = nullptr;
+    }
+
+    MapObject::Get_World_Dict()->Clear();
 }
 
-bool WorldHeightMap::Get_Flip_State(int x, int y)
+bool WorldHeightMap::Get_Flip_State(int x, int y) const
 {
-#ifdef GAME_DLL
-    return Call_Method<bool, WorldHeightMap, int, int>(PICK_ADDRESS(0x007449F0, 0x005EBE61), this, x, y);
-#else
+    if (x < 0 || y < 0) {
+        return false;
+    }
+
+    if (y >= m_height) {
+        return false;
+    }
+
+    if (x >= m_width) {
+        return false;
+    }
+
+    if (m_cellFlipState) {
+        return ((1 << (x & 7)) & m_cellFlipState[(x / 8) + m_flipStateWidth * y]) != 0;
+    }
+
     return false;
-#endif
+}
+
+void WorldHeightMap::Set_Flip_State(int x_index, int y_index, bool state)
+{
+    if (x_index >= 0 && y_index >= 0 && y_index < m_height && x_index < m_width) {
+        if (m_cellFlipState) {
+            unsigned char *c = &m_cellFlipState[(x_index / 8) + m_flipStateWidth * y_index];
+            if (state) {
+                *c |= 1 << (x_index & 7);
+            } else {
+                *c &= ~(unsigned char)(1 << (x_index & 7));
+            }
+        }
+    }
+}
+
+void WorldHeightMap::Clear_Flip_States()
+{
+    if (m_cellFlipState) {
+        memset(m_cellFlipState, 0, m_height * m_flipStateWidth);
+    }
+}
+
+float WorldHeightMap::Get_Bilinear_Sample_Seismic_Z_Velocity(int x, int y)

```

Comment:
whats going on in this function? every - 1 + x and + 1 + x is inverse of what pseudo shows, for me at least it shows
`m_seismicZVelocities[m_width * (y - 1) + x - 1];`
`m_seismicZVelocities[m_width * (y + 1) + x + 1];`